File: 8Kyu/remove.js

//#Description:
//Remove an exclamation mark from the end of a string. For a beginner kata, you can assume that the input data is always a string, no need to verify it.

/*Examples*/
/*remove("Hi!") === "Hi"*/
/*remove("Hi!!!") === "Hi!!"*/
/*remove("!Hi") === "!Hi"*/
/*remove("!Hi!") === "!Hi"*/
/*remove("Hi! Hi!") === "Hi! Hi"*/
/*remove("Hi") === "Hi"*/


function remove(s) {
  return s.endsWith('!') ? s.slice(0, -1) : s;
}
function remove (string) {
  //coding and coding....
  
  let str = string.split('').map(a=> a=='!'?'':a).join('')
  return str+'!'
  }

File: 8Kyu/remove.py
def remove(s):
    i = -1
    while s[i] == '!':
        i-=1
    if(i == -1):
        return s
    return s[:i+1]
        

# Description:
# Remove all exclamation marks from the end of sentence.

# Examples
# remove("Hi!") === "Hi"
# remove("Hi!!!") === "Hi"
# remove("!Hi") === "!Hi"
# remove("!Hi!") === "!Hi"
# remove("Hi! Hi!") === "Hi! Hi"
# remove("Hi") === "Hi"
